src/sentry/snuba/metrics/fields/base.py

```diff
@@ -245,8 +247,18 @@ def __recursively_get_all_entities_in_derived_metric_dependency_tree(
         if derived_metric_name not in DERIVED_METRICS:
             metric_type = get_single_metric_info(projects, derived_metric_name)["type"]
             return {METRIC_TYPE_TO_ENTITY[metric_type].value}
+
         entities = set()
         derived_metric = DERIVED_METRICS[derived_metric_name]
+
+        try:
+            # Stop the recursion if the entity for this derived metric was computed previously
+            entity = derived_metric.get_entity()
+            if derived_metric.get_entity() is not None:
+                return {entity}
+        except DerivedMetricParseException:
+            ...
```

Comment:
It seems we do this throughout the file... can we just use `pass`?

Reply:
why?

Reply:
I don't think this is a common pattern to use, but I will follow up with you on slack